cpp/src/parquet/metadata.cc

```diff
@@ -47,6 +47,17 @@ const ApplicationVersion& ApplicationVersion::PARQUET_CPP_FIXED_STATS_VERSION()
   return version;
 }
 
+std::string ParquetVersionToString(ParquetVersion::type ver) {
+  switch (ver) {
+    case ParquetVersion::PARQUET_1_0:
+      return "1.0";
+    case ParquetVersion::PARQUET_2_0:
+      return "2.0";
+    default:
+      return "UNKNOWN";
```

Comment:
If you remove the default case, and move the `return "UNKNOWN"` outside the switch, then `-Wall` should imply `-Wswitch` which would catch at compile time when a newer version isn't handled while still working with older `parquet-reader` reading newer parquet files.

The goal is that when a new enum is added, we don't forget to update this function. (Using `-Wswitch-enum` is a bit radical).